### PR TITLE
feat(sync): Display the number of connected peers in the interface.

### PR DESCRIPTION
**Pull request by almet**
Display the number of connected peers in the interface if the websocket mode is activated on this map.

image

### CODE REVIEW

File: umap/static/umap/js/modules/schema.js
Comment (yohanboniface) [inline umap/static/umap/js/modules/schema.js:316]: I'm not convinced this has to be in the schema here, given it's not a stored property. I'd simply have a method that updates this counter (with a css class accessor or something), wdyt ?

File: umap/static/umap/js/umap.controls.js
Comment (almet) [reply umap/static/umap/js/umap.controls.js:626]: Oh, you mean that I could have just used this `update` function to put the changes in it! It didn't cross my mind at the time, but I will definitely do it!

File: umap/static/umap/js/umap.controls.js
Comment (almet) [reply umap/static/umap/js/umap.controls.js:666]: Ah, I just copy/pasted from another call tbh. Will update 👍

File: umap/static/umap/map.css
Comment (almet) [reply umap/static/umap/map.css:518]: Sure thing!

File: umap/static/umap/js/modules/schema.js
Comment (almet) [reply umap/static/umap/js/modules/schema.js:316]: I was wondering the same thing when writing it. Using the schema is mainly a way to tell what part of the interface needs to be updated. If we weren't to use the schema, we would special-case this in the `render()` method of the map.

I don't think there is a clear winner between the two ways here, I believe we just have to decide on one and stick to it for the other cases as well. On one case we have to check not to forward these events to other peers, and on the other we have to call the part that updates the interface after an if branch testing for this specific property.

Comment (almet) [reply umap/static/umap/js/modules/schema.js:316]: I'll let you decide @yohanboniface 👍

File: umap/static/umap/js/umap.controls.js
Comment (yohanboniface) [inline umap/static/umap/js/umap.controls.js:660]: I don't get where is this method defined, can you point me to it , :)

File: umap/static/umap/js/modules/schema.js
Comment (yohanboniface) [reply umap/static/umap/js/modules/schema.js:316]: I'm trying to get the full picture so to have an opinion on this.
Two elements:
- I'm expecting the schema to reflect the actual data, so if we add `numberOfConnectedPeers` to the schema, it means it's somewhere on the map object (I guess in `options` like others for now)
- maybe we can keep this property in the schema, and add a `sync` property to schema entries, defaulting to true, but when false we do not sync changes to other peers ?

File: umap/static/umap/js/umap.controls.js
Comment (almet) [reply umap/static/umap/js/umap.controls.js]: The `update` function has been removed in favor of `umap.propagate` in 0d7c6e451d0440304b3c55a799670c27ded81e9c, so we're good here.

File: umap/static/umap/js/umap.js
Comment (almet) [reply umap/static/umap/js/umap.js:228]: Removed :+1:

File: umap/static/umap/js/umap.controls.js
Comment (almet) [reply umap/static/umap/js/umap.controls.js]: Updated in 1cce90f990a797fc03d931e24215113d6e332f08

File: umap/static/umap/map.css
Comment (almet) [reply umap/static/umap/map.css]: Updated in 0228c787fe60acd4359e821be8c1a08e5159b5bb

Comment (almet) [reply umap/static/umap/map.css]: I actually did some more changes there by adding a `--color-dark` variable. Let me know if that's good for you :-)

File: umap/static/umap/js/umap.controls.js
Comment (almet) [reply umap/static/umap/js/umap.controls.js]: It's currently not implemented, I removed it for now, it can come later on.

**Comment by almet** [comment]
I'm not sure why, but when on this branch, map name and visibility is not displayed when I load a map. It is once I edit the title, though.

I don't understand how the changes I did here interact with this part of the code. Could you perhaps give a look @yohanboniface ?

**Comment by almet** [comment]
> I'm not sure why, but when on this branch, map name and visibility is not displayed when I load a map. It is once I edit the title, though.

I believe we're hitting https://github.com/umap-project/umap/issues/2225 (I created a new issue for this), and that's it's in fact not related with the changes here.

We should be good to merge once I have your green light @yohanboniface!

**Comment by almet** [comment]
> I believe we're hitting https://github.com/umap-project/umap/issues/2225 (I created a new issue for this), and that's it's in fact not related with the changes here.

And actually it **was** a problem with this PR. After calling `renderToolbar`, one need to call `this.propagate()` to populate back the name and permission fields on the toolbar.

We're considering changing this so that it's done automatically, but I've updated the code to do this for now.

**Review by yohanboniface — APPROVED** [review]

**Pull request merged**
